src/rgw/driver/rados/rgw_sal_rados.cc

```diff
@@ -1670,7 +1671,12 @@ int RadosObject::modify_obj_attrs(const char* attr_name, bufferlist& attr_val, o
   state.obj = target;
   set_atomic();
   state.attrset[attr_name] = attr_val;
-  r = set_obj_attrs(dpp, &state.attrset, nullptr, y);
+
+  // make a tiny adjustment to the existing mtime so that fetch_remote_obj()
+  // won't return ERR_NOT_MODIFIED when syncing the modified object
+  auto mtime = state.mtime + std::chrono::nanoseconds(1);
+
+  r = set_obj_attrs(dpp, &state.attrset, nullptr, y, mtime);
```

Comment:
in rgw_op.cc, i see that some ops call `s->object->modify_obj_attrs()` while others call `set_obj_attrs()`. i think we'll want all of those changes to replicate

since `RadosObject::set_obj_attrs()` covers all of those, it's probably a better place to override the mtime. 
and if this is internal to `set_obj_attrs()`, we shouldn't have to change the sal api at all

Reply:
@cbodley I initially thought about making the changes in set_obj_attrs() but there are some places in sal_posix like POSIXObject::copy() and POSIXMultipartUpload::init() where set_obj_attrs() are called. i wasn't sure if changing the mtime was desirable there.

Reply:
changing the mtime in `RadosObject::set_obj_attrs()` wouldn't have any effect on the posix store. eventually other backends will want to handle this too

Reply:
done